### DISCUSSION TITLE
Disabling and Enabling  Serial by tempering with registers/pins after waking-up or before sleeping?

### DISCUSSION THREAD

**Discussion by dattasaurabh82** [SpenceKonde/megaTinyCore#509] [Library discussion]
So I have a situation. _Like always_ 😅

### Setup:
**uC:** ATTINY1607-MFR clocked at 5MHz (necessary Freq for my main application later)
**Operating Voltage:** 3.7V from 120mAH LiPo Battery.
BOD disabled in Fuse (From IDE's tools Menu)
`millis()`, `micros()`, enabled and are in use 

  Some UART functionalities are required.
  So some pins are always connected to an FTDI board as such:

```
 LIPO VCC <-> [VCC] UC      
 LIPO GND <-> [GND] uC <--> FTDI GND <-> USB GND
                            FTDI VCC <-> USB VCC
              [RXD] PB3 <-- FTDI TXD
              [TXD] PB2 --> FTDI RXD                      
```

### Problem:
I need to use serial data, when the device is Awake. After that the uC sys goes to sleep. 
Preferably the FTDI serial chip should be in one PCB with the uC. If we remove it from the uC's UART bus physically detaching it from RX, TX and GND pins, then the current consumption is 0.6uA by the system. with it it is 11mA during sleep.  
 
### Current Code struct (minimal here for problem explanation):
```
#include <avr/sleep.h>

void setup(){
   // disabled unused pins
   // enabled a btn to a pin in INPUT_PULLUP for async interrupt used to wake up the uC from sleeping.
   // some other things... 

   // Sleep mode init
   set_sleep_mode(SLEEP_MODE_PWR_DOWN);
   sleep_enable();
} 

void loop(){
  // when the sys is awake, 
  // only once enable Serial
  Serial.begin(115200);

  // also keep an eye on if data is coming through serial. 
  // if so do something with it ...
  // Anyways Do something ...

  // Then go to sleep again
  Serial.end();
  sleep_cpu();
}
```
**Sleep mode current:** _~116uA_  which is lower than ~11mA earlier before I used this mechanism. 
It is still some 200 time more consumption compared to the current consumption when the FTDI chip is physically removed from the uC's UART bus (_As then the current consumption is: 0.5uA_).

### Tried to set TX pin to input pull-up before going to sleep: 
Basically can we do something with the TX Pin and some other register (if I'm missing something..)

Also tried
```
Serial.end();
// ... ? what else can we do here to sever the RX pin ? 
sleep_cpu();
```
1607's TX pin when constantly connected to FTDI's RX PIN, has not affect. 
1607's RX pin when separated from FTDI's TX PIN reduces current consumption drastically. 
**How can we Reduce it further without removing the FTDi chip from the uC and it's UART bus?** 

[https://onlinedocs.microchip.com/pr/GUID-C35EC6FB-C2A0-42FF-80EF-BF1E9A5B3EE8-en-US-2/index.html?GUID-EC8461C8-689A-4FF2-A949-48CCCD0E192A](https://onlinedocs.microchip.com/pr/GUID-C35EC6FB-C2A0-42FF-80EF-BF1E9A5B3EE8-en-US-2/index.html?GUID-EC8461C8-689A-4FF2-A949-48CCCD0E192A)

### Update:
I managed to reduce the power by putting the RX pin (which is usually pulled high) to OUTPUT and setting LOW. 
```
Serial.end();
pinMode(8, OUTPUT);
digitalWrite(8, LOW);
sleep_cpu();
```
Don't know if that's the correct way to do it though...

**Comment by SpenceKonde** [comment]
I think you can also just set the pin INPUT.

And wait, so it doesn't burn power if the tiny's TX pin is connected to FTDI's RX? o_o 

And wait, 11mA or 110uA? 110uA is what I would expect if the RX line were tied to an unpowered device, because the pullup is enabled,, but the pin is connected to something that is approximately grounded (within the low current regime, it might as well be tied to ground).11mA is a whole different ball game....
